panther_lights/src/controller_node.py

```diff
@@ -233,55 +375,36 @@ def _get_animation_by_id(self, animation_id: int) -> PantherAnimation:
                         'Missing \'both\' or \'front\'/\'rear\' in animation description'
                     )
 
-                if 'interrupting' in anim:
-                    animation.interrupting = anim['interrupting']
-
-                return animation
-
-        raise KeyError(f'No Animation with id: {animation_id}')
-
-    def _add_animation_to_queue(self, animation: PantherAnimation):
-        self._interrupt = animation.interrupting
-        if self._interrupt:
-            self._anim_queue.queue.insert(0, animation)
-        else:
-            self._anim_queue.put(animation)
-        if animation.repeating:
-            self._default_animation = animation
-
-    def _get_image_animation_description(self, animation: LEDImageAnimation):
-        if not animation.image:
-            raise Exception('missing required field \'image\'')
-
-        animation_description = {
-            'image': animation.image,
-            'duration': animation.duration,
-            'repeat': animation.repeat,
-            'brightness': animation.brightness,
-        }
-
-        if animation.color:
-            animation_description.update({'color': animation.color})
-
-        return animation_description
+                if 'name' in anim:
+                    animation.name = anim['name']
+
+                if 'priority' in anim:
+                    priority = anim['priority']
+                    if (
+                        not 0 < priority <= PantherAnimation.ANIMATION_DEFAULT_PRIORITY
+                        or not isinstance(priority, int)
+                    ):
+                        priority = PantherAnimation.ANIMATION_DEFAULT_PRIORITY
+                        rospy.logwarn(
+                            f'{rospy.get_name()} Invalid priority for animaiton: {animation.name}. Using default'
+                        )
+                    animation.priority = priority
 
-    def _update_animations(self):
+                if 'timeout' in anim:
+                    timeout = anim['timeout']
+                    if timeout <= 0:
+                        rospy.logwarn(
+                            f'{rospy.get_name} Invalid timeout for animation: {animation.name}. Using default'
+                        )
+                        timeout = PantherAnimation.ANIMATION_DEFAULT_TIMEOUT
+                    animation.timeout = timeout
 
-        user_animations = rospy.get_param('~user_animations', '')
+                animations.append(dict(id=anim['id'], animation=animation))
```

Comment:
```suggestion
                animations[anim['id']] = animation
```
I meant something like this